src/compiler/interface.jl

```diff
@@ -250,7 +250,7 @@ function pullback(f, ps::Params)
       cache(cx)[p] = nothing
     end
     back(Δ)
-    Grads(cx.cache, ps) # TODO make a copy
+    Grads(IdDict(k => _unthunk(v) for (k, v) in cx.cache), ps) # TODO make a copy
```

Comment:
this is the bit that it is struggled to handle in nested AD I think.
Possibly try (just for now)
```suggestion
    Grads(IdDict([k => _unthunk(v) for (k, v) in cx.cache]), ps) # TODO make a copy
```

Reply:
Thanks, I'll do a quick check regarding type inference of the one vs. the other suggested version.